airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -0,0 +1,189 @@
+from abc import ABC, abstractmethod
+from typing import Any, Tuple, Mapping, Iterable, Optional
+
+from base_python import ApiTokenAuthenticator, AbstractSource, HttpStream, IncrementalStream, Stream
+
+
+class StripeStream(HttpStream):
+    url_base = "https://api.stripe.com/v1/"
+
+    def get_next_page_token(self, decoded_response: Mapping[str, Any]) -> Optional[Mapping[str, Any]]:
+        if decoded_response.get('has_more') and bool(decoded_response['has_more']):
+            if decoded_response.get('data') and len(decoded_response['data']) > 0:
+                last_object_id = decoded_response['data'][-1]['id']
+                return {'starting_after': last_object_id}
+
+    def parse_response(self, decoded_response: Mapping) -> Iterable[Mapping]:
+        if decoded_response.get('data'):
+            yield from decoded_response['data']
+
+
+class IncrementalStripeStream(StripeStream, IncrementalStream, ABC):
```

Comment:
This doesn't look great. let's discuss it offline

Reply:
merged IncrementalStream with Stream